lib/templates/android-kotlin/inc/_attributes_properties.liquid

```diff
@@ -0,0 +1,60 @@
+{% for attribute in entity.attributes -%}
+  {%- if attribute.realm_read_only.size == 0 -%}
+      {%- comment %} ******* CONVERT TYPE CAPTURE ******* {% endcomment -%}
+      {%- capture convert_type -%}
+        {%- if attribute.realm_read_only.size == 0 and attribute.enum_type.size > 0 -%}
+          String
+        {%- else -%}
+          {%- include 'inc/type_converter' -%} 
+        {%- endif -%}
+      {%- endcapture -%}
+
+      {%- capture is_primitive %}
+        {%- if attribute.realm_read_only.size == 0 and attribute.enum_type.size > 0 -%}
+          false
+        {%- else -%}
+          {%- include 'inc/type_primitives' -%} 
+        {%- endif -%}
+      {%- endcapture -%}
+
+      {%- capture default_value %}
+        {%- if attribute.default.size > 0 %} =
+          {%- if convert_type == "Boolean" %}
+            {%- if attribute.default == "YES" %} true {%- else %} false {%- endif %}
+          {%- elsif convert_type == "String" %} "{{ attribute.default }}"
+          {%- else %} {{ attribute.default }}
+          {%- endif %}
+        {%- elsif attribute.enum_type.size > 0 %} = 
+          {%- if attribute.optional == true %} null
+          {%- else %} "{{ attribute.enum_values.first }}"
+          {%- endif -%}
+        {%- else %} = 
+          {%- if attribute.optional == true and is_primitive == "false" %} null 
+          {%- else %} {% include 'inc/type_defaults' -%}
+          {%- endif %}
+        {%- endif %}
+      {%- endcapture %}
+
+      {%- capture nullable_type %}
+        {%- if attribute.optional == true %}?{%- endif -%}
+      {%- endcapture %}
+
+      {%- assign name = attribute.name -%}
+      {%- if name == primary_key %}
+    @PrimaryKey 
+      {%- endif -%}
+      {%- if attribute.optional == false and attribute.realm_ignored == false and name != primary_key and is_primitive == "false" %}
+    @Required 
```

Comment:
I'm wondering if that's really necessary. I mean, we didn't have it in the Java template (and maybe we should), but in Kotlin if a field is e.g. a `Double` then it will always have a value, so mentioning `@Required` here is kind of redondant? And if it is a `Double?` then it doesn't make sense to have `@Required`?

In short, if in the xcdatamodel you check the "Optional" checkbox, you'll have optional types (with `?` postfix) AND no `@Required`, and if you uncheck the box you'll have no `?` AND `@Required`… so that seems redondant to me.

That being said, maybe that doesn't hurt to have them (I'm no Android dev so no expert here), but in that case maybe we should add them to the Java templates too?

Reply:
It's true that Kotlin adds more strict nullability information, but the `@Required` annotation tells the Realm engine to perform additional checks, just in case. In my opinion, better safe than sorry.

Reply:
Ok. Agreed. But that rationale holds true for Java too I guess, so it would be cool to add those `@Required` to the java template too. Probably for a separate, dedicated PR though.